core/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java

```diff
@@ -207,6 +211,24 @@ void execute(Terminal terminal, String pluginId, boolean isBatch, Map<String, St
             Files.createDirectory(env.pluginsFile());
         }
 
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            for (final Path pathToDeleteOnShutdown : pathsToDeleteOnShutdown) {
+                try {
+                    IOUtils.rm(pathToDeleteOnShutdown);
+                } catch (final IOException e) {
+                    try (
+                        final StringWriter sw = new StringWriter();
+                        final PrintWriter pw = new PrintWriter(sw)) {
+                        e.printStackTrace(pw);
+                        terminal.println(sw.toString());
+                    } catch (final IOException impossible) {
+                        // StringWriter#close declared a checked IOException from the Closeable interface but the Javadocs for StringWriter
```

Comment:
Should we still log something here, in case `terminal.println` throws an IOException? Or the impossible becomes possible (hey, you never know with JDK9)